jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletChannel.java

```diff
@@ -794,19 +810,25 @@ public void onCompleted()
         {
             Authentication authentication = apiRequest.getAuthentication();
             if (authentication instanceof Authentication.User userAuthentication)
-                _request.setAttribute(CustomRequestLog.USER_NAME, userAuthentication.getUserIdentity().getUserPrincipal().getName());
+                _servletContextRequest.setAttribute(CustomRequestLog.USER_NAME, userAuthentication.getUserIdentity().getUserPrincipal().getName());
 
-            String realPath = apiRequest.getServletContext().getRealPath(Request.getPathInContext(_request));
-            _request.setAttribute(CustomRequestLog.REAL_PATH, realPath);
+            String realPath = apiRequest.getServletContext().getRealPath(Request.getPathInContext(_servletContextRequest));
+            _servletContextRequest.setAttribute(CustomRequestLog.REAL_PATH, realPath);
 
-            String servletName = _request.getServletName();
-            _request.setAttribute(CustomRequestLog.HANDLER_NAME, servletName);
+            String servletName = _servletContextRequest.getServletName();
+            _servletContextRequest.setAttribute(CustomRequestLog.HANDLER_NAME, servletName);
         }
 
         // Callback will either be succeeded here or failed in abort().
+        Callback callback = _callback;
+        // Must recycle before notification to allow for reuse.
+        // Recycle always done here even if an abort is called.
+        recycle();
         if (_state.completeResponse())
-            _callback.succeeded();
-        _combinedListener.onComplete(_request);
+        {
+            _combinedListener.onComplete(_servletContextRequest);
```

Comment:
If line 826 calls `recycle()`, won't the `_servletContextRequest` at line 829 always be null for the call to `onComplete()`?